component/discovery/process/process.go

```diff
@@ -0,0 +1,83 @@
+package process
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.process",
+		Args:    Arguments{},
+		Exports: discovery.Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+func New(opts component.Options, args Arguments) (*Component, error) {
+	c := &Component{
+		l:             opts.Logger,
+		onStateChange: opts.OnStateChange,
+		argsUpdates:   make(chan Arguments),
+		args:          args,
+	}
+	return c, nil
+}
+
+type Component struct {
+	l             log.Logger
+	onStateChange func(e component.Exports)
+	processes     []discovery.Target
+	argsUpdates   chan Arguments
+	args          Arguments
+}
+
+func (c *Component) Run(ctx context.Context) error {
+	doDiscover := func() error {
+		processes, err := discover(c.l, &c.args.DiscoverConfig)
+		if err != nil {
+			return err
+		}
+		c.processes = convertProcesses(processes)
+		c.changed()
+		return nil
+	}
+	if err := doDiscover(); err != nil {
+		return err
+	}
+
+	t := time.NewTicker(c.args.RefreshInterval)
```

Comment:
```suggestion
	t := time.NewTicker(c.args.RefreshInterval)
	defer t.Stop()
```

Reply:
FYI here this is not a big deal, but it's good habit to release ticker and timer since otherwise they leak resource.